src/store/modules/events/index.ts

```diff
@@ -494,27 +499,64 @@ const module: Module<EventsModuleState, RootState> = {
       const skip = (state.repetitions[key] || []).length;
 
       const response = await eventsApi.getLatestRepetitions(projectId, eventId, skip, limit);
-      const repetitions = response.data.project.event.repetitions;
-
-      /**
-       * Solution for not displaying both `userAgent` and `beautifiedUserAgent` addons
-       */
-      filterBeautifiedAddons(repetitions);
 
       if (originalEvent) {
         filterBeautifiedAddons([ originalEvent ]);
       }
 
-      repetitions.map(repetition => {
-        // save to the state
-        commit(MutationTypes.AddRepetitionPayload, {
-          projectId,
-          eventId,
-          repetition: {
+      /**
+       * If delta is present, apply delta to the event payload
+       */
+      const repetitions = response.data.project.event.repetitions.map(repetition => {
+        let processedRepetition: HawkEventRepetition;
+
+        const isNewDeltaFormat = !repetition.payload;
```

Comment:
maybe it would be better to check for `delta` existence?

Reply:
delta can be empty in case of not diff between original event and repetition